rules/legacy/src/Rector/Include_/AddTopIncludeRector.php

```diff
@@ -181,20 +154,48 @@ private function initialize(): void
         $reflection = new ReflectionClass(Include_::class);
         $constants = $reflection->getConstants();
 
-        if (! isset($constants[$this->configuration['type'] ?? ''])) {
-            throw new InvalidRectorConfigurationException('Invalid type: must be one of ' . implode(
+        if (! isset($constants[$this->settings['type'] ?? ''])) {
+            throw new InvalidRectorsettingsException('Invalid type: must be one of ' . implode(
                 ', ',
                 array_keys($constants)
             ));
         }
 
-        if ('' === ($this->configuration['file'] ?? '')) {
-            throw new InvalidRectorConfigurationException('Invalid parameter: file must be provided');
+        if ('' === ($this->settings['file'] ?? '')) {
```

Comment:
3 operations at one line. This is bit confusing to read... what is the goal of this ?

My guess: 

```suggestion
        if (! isset($this->settings['file'])) {
```

Reply:
I originally had empty($this->settings['file']) but PHPStan complained "Use strict comparison instead of empty".  empty is actually correct in this situation. If the index is not set, it is an error and empty.  If the index is set, but empty ('', 0, etc), it is also an error.  It needs to be a string of some sort.  isset would need an additional test.

So I went with the null coalesce operator, but you have to wrap it in () to make sure the operator precedence is always correct, then type compare it to string to make sure it is not empty.  I could reverse the lhs and rhs, but it is my style to put the scalar constant on the lhs so if you mistype a comparison as as assign, you get an error. I have noticed your style guide does not do this.  I would recommend it.

Other suggestions?

Reply:
I wonder how to make it simple to prevent such a rocket engeneering over one variable.

What about makint it a parameter? The file name is required anyway, right?

Reply:
See: https://github.com/rectorphp/rector/pull/3688#discussion_r454419900